packages/gestalt/src/SideNavigationGroup.js

```diff
@@ -0,0 +1,256 @@
+  expanded: boolean,
+  selectedItemId: string,
+  itemId: string,
+  paddingStyle: {|
+    paddingInlineStart: string | number | void,
+    paddingInlineEnd: string | number | void,
+  |},
+  icon?: IconType,
+  label: string,
+  badge?: BadgeType,
+  notificationAccessibilityLabel?: string,
+  counter?: Counter,
+  display?: Display,
+|}) {
+  return (
+    <Box
+      color={itemColor ?? undefined}
+      paddingY={2}
+      minHeight={44}
+      rounding={2}
+      display="flex"
+      alignItems="center"
+      dangerouslySetInlineStyle={{
+        __style:
+          expanded && selectedItemId === itemId
+            ? {
+                border: `2px solid var(--color-background-selected-base)`,
+                ...paddingStyle,
+              }
+            : paddingStyle,
+      }}
+    >
+      <Flex gap={2} height="100%" width="100%">
+        {icon ? (
+          <Flex.Item alignSelf="center">
+            <Box aria-hidden>
+              {typeof icon === 'string' ? (
+                <Icon inline icon={icon} accessibilityLabel="" color="default" />
+              ) : (
+                <Icon inline dangerouslySetSvgPath={icon} accessibilityLabel="" color="default" />
+              )}
+            </Box>
+          </Flex.Item>
+        ) : null}
+        <Flex.Item alignSelf="center" flex="grow">
+          <Text inline color="default">
+            {label}
+            {(badge || notificationAccessibilityLabel) && (
+              <Box marginStart={1} display="inlineBlock" height="100%">
+                {/* Adds a pause for screen reader users between the text content */}
+                <Box display="visuallyHidden">{`, `}</Box>
+                {!notificationAccessibilityLabel && badge ? (
+                  <Badge text={badge.text} type={badge.type} />
+                ) : null}
+                {notificationAccessibilityLabel ? (
+                  <Box
+                    aria-label={notificationAccessibilityLabel}
+                    height={8}
+                    width={8}
+                    rounding="circle"
+                    color="primary"
+                    role="status"
+                  />
+                ) : null}
+              </Box>
+            )}
+          </Text>
+        </Flex.Item>
+        {counter && (
+          <Flex.Item flex="none" alignSelf="center">
+            <Box display="visuallyHidden">{`, `}</Box>
+            <Box
+              aria-label={counter.accessibilityLabel}
+              role="status"
+              marginEnd={display === 'static' ? -2 : undefined}
+            >
+              <Text align="end" color="default">
+                {counter.number}
+              </Text>
+            </Box>
+          </Flex.Item>
+        )}
+        {display === 'expandable' ? (
+          <Flex.Item flex="none" alignSelf="center">
+            {/* marginEnd={-2} is a hack to correctly position the counter as Flex + gap + width="100%" doean't expand to full width */}
```

Comment:
I tried a lot here. I put together the Flex training so I had a good understanding of the existing possibilities and caveats. The issue here is that when I remove marginEnd={-2} and remove width="100%" from the parent to get the correct spacing then another design spec gets disaligned. Then flex="grow" breaks. And it breaks because we are setting a height in Box that's needed to self center the contents. 

I'd ship like this and maybe revisit it with fresh eyes. I spent an entire evening and finally marginEnd={-2}  came to my mind as the only salvation.

Reply:
hahah I trust you, was just curious cuz I noticed the same on SlimBanner - this looks good!